cloudconfig/podcfg/podcfg_test.go

```diff
@@ -98,3 +98,25 @@ func (*podcfgSuite) TestBootstrapConstraints(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(podConfig.Bootstrap.BootstrapMachineConstraints, gc.DeepEquals, cons)
 }
+
+func (*podcfgSuite) TestFinishControllerPodConfig(c *gc.C) {
+	cfg := testing.CustomModelConfig(c, testing.Attrs{
+		"type": "kubernetes",
+	})
+	podConfig, err := podcfg.NewBootstrapControllerPodConfig(
+		testing.FakeControllerConfig(),
+		"controller-1",
+		"kubernetes",
+		constraints.Value{},
+	)
+	podcfg.FinishControllerPodConfig(
+		podConfig,
+		cfg,
+		map[string]string{"foo": "bar"},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(podConfig.AgentEnvironment, jc.DeepEquals, map[string]string{
+		"PROVIDER_TYPE": "kubernetes",
```

Comment:
I'm not sure we actually use the PROVIDER_TYPE anywhere any more, and it could be removed I think. Worth checking though, but it used to provide custom networking stuff for maas, but shouldn't any more. Not for this branch, but would be a good cleanup.

Reply:
It's currently used for agent start up to detect if it's a k8s model or not.
The original use is not there anymore.
With a fair bit of refactoring, we can migrate away from it as controller/machine agent separation is done etc.